File: hackerrank/easy/0029-manasa-and-stones/js/solution.mjs
// solution.mjs

export function stones(n, a, b) {
  if (a === b) return [(n - 1) * a];
  const low = Math.min(a, b), high = Math.max(a, b);
  const res = [];
  for (let k = 0; k < n; k++) {
    res.push((n - 1 - k) * low + k * high);
  }
  return res;
}

// --- demo run (from samples) ---
console.log(stones(3, 1, 2));      // [2, 3, 4]
console.log(stones(4, 10, 100));   // [30, 120, 210, 300]

File: hackerrank/easy/0029-manasa-and-stones/python/solution.py
## solution.py

from typing import List

def stones(n: int, a: int, b: int) -> List[int]:
    if a == b:
        return [(n - 1) * a]
    low, high = min(a, b), max(a, b)
    return [(n - 1 - k) * low + k * high for k in range(n)]


# --- demo run (from samples) ---
if __name__ == "__main__":
    print(stones(3, 1, 2))       # [2, 3, 4]
    print(stones(4, 10, 100))    # [30, 120, 210, 300]